core/utils/wei.go

```diff
@@ -198,3 +203,73 @@ func (w *Wei) UnmarshalText(b []byte) error {
 	}
 	return errors.Errorf("unable to parse %q", s)
 }
+
+func (w *Wei) ToInt() *big.Int {
+	return (*big.Int)(w)
+}
+
+func (w *Wei) Int64() int64 {
+	return w.ToInt().Int64()
+}
+
+func (w *Wei) Cmp(y *Wei) int {
+	return w.ToInt().Cmp(y.ToInt())
+}
+
+func (w *Wei) IsNegative() bool {
+	return w.Cmp(NewWeiI(0)) < 0
+}
+
+func (w *Wei) Equal(y *Wei) bool {
+	return w.Cmp(y) == 0
+}
+
+func WeiMax(x, y *Wei) *Wei {
+	return NewWei(bigmath.Max(x.ToInt(), y.ToInt()))
+}
+
+func WeiMin(x, y *Wei) *Wei {
+	return NewWei(bigmath.Min(x.ToInt(), y.ToInt()))
+}
+
+// NOTE: Maths functions always return newly allocated number and do not mutate
+
+func (w *Wei) Sub(y *Wei) *Wei {
+	result := big.NewInt(0).Sub(w.ToInt(), y.ToInt())
+	return NewWei(result)
+}
+
+func (w *Wei) Add(y *Wei) *Wei {
+	return NewWei(big.NewInt(0).Add(w.ToInt(), y.ToInt()))
+}
+
+func (w *Wei) Mul(y *big.Int) *Wei {
+	return NewWei(big.NewInt(0).Mul(w.ToInt(), y))
+}
+
+func (w *Wei) IncreaseByPercentage(percentage uint32) *Wei {
+	bumped := new(big.Int)
+	bumped.Set(w.ToInt())
+	bumped.Mul(w.ToInt(), big.NewInt(int64(100+percentage)))
+	bumped.Div(bumped, big.NewInt(100))
+	return NewWei(bumped)
+}
+
+// Returns whichever is greater, the percentage bump or the bump by fixed increment
+func (w *Wei) IncreaseByPercentageOrIncrement(percentage uint32, increment *Wei) (bumped *Wei) {
```

Comment:
If `Scale` makes sense above, then maybe something like this?
```suggestion
func (w *Wei) MaxScaleOrAdd(percentage uint32, add *Wei) (bumped *Wei) {
```